samza-core/src/main/java/org/apache/samza/clustermanager/SamzaResourceRequest.java

```diff
@@ -109,15 +135,24 @@ public String toString() {
             ", requestId='" + requestId + '\'' +
             ", processorId=" + processorId +
             ", requestTimestampMs=" + requestTimestamp +
+            ", faultDomains=" + convertFaultDomainSetToString() +
             '}';
   }
 
-  /**
-   * Requests are ordered by the processor type and the time at which they were created.
-   * Requests with timestamps in the future for retries take less precedence than timestamps in the past or current.
-   * Otherwise, active processors take precedence over standby processors, regardless of timestamp.
-   * @param o the other
-   */
+  private String convertFaultDomainSetToString() {
+    StringBuilder faultDomainSb = new StringBuilder();
```

Comment:
just curious, wont it work without this -esp since you already defined the "toString" for FaultDomain?

Reply:
Yes it should. I had added the FaultDomain toString method later on hence forgot to remove this. Thanks for observing :)